### DISCUSSION TITLE
Is Fine Offset WS69 weather station supported?

### DISCUSSION THREAD

**Discussion by mcmlxxvii** [merbanan/rtl_433#3059] [Q&A] [answered]
I see that there's a registering protocol [213] already in place for "_Fine Offset Electronics WS80 weather station_" and [113] for "_Ambient Weather WH31E Thermo-Hygrometer Sensor, EcoWitt WH40 rain gauge, WS68 weather station_", so I'm wondering: Is [Ecowitt WS69](https://shop.ecowitt.com/en-ca/products/ws69) also supported by rtl_433?

If yes, which rtl_433 protocol ID should I use? 

If not, are there any plans to support it in the near future?

Thanks!

**Comment by zuckschwerdt** [comment]
Support for the WS69 was asked before but I don't know if anyone took a look at it. It should be very similar to others we already support.
https://groups.google.com/g/rtl_433/c/aFjgg-4kdJ8/m/OjCaIrg6AAAJ

**Comment by tomburke25** [comment]
Mine seems to identify as a WH65B (so -R 78) 

I've only just starting messing with it, but sensor data seems to be correct

**Reply by neillbell** [reply]
I was looking at the WS69 today, so it's surprising to find that you posted about this just last week.
Were you able to make any progress?

**Reply by kbahey** [reply]
@tomburke25 

So all the sensor values show up (rain, wind, solar, ...etc.)?

**Reply by sillyfrog** [reply]
I have a [LILYGO® TTGO LoRa32 V2.1_1.6](https://www.aliexpress.com/item/32872078587.html) board, running [OpenMQTTGateway](https://github.com/1technophile/OpenMQTTGateway), I used the pre-built `lilygo-rtl_433-fsk` image - (note: After flashing, I had to set the frequency to 433.920 MHz in the web UI, as this didn't happen by default).

I have the [Ecowitt WS80](https://www.ecowitt.com/shop/goodsDetail/46#), and the [Ecowitt WS69](https://shop.ecowitt.com/en-au/products/ws69?variant=41418885169314), and both "just work" with the above MQTT Gateway. I did not have to do anything to configure the weather station once MQTT was setup to get the data into Home Assistant, all the discovery MQTT events were sent and the data started coming in.

Sample MQTT event from the WS80:
```
.../RTL_433toMQTT/Fineoffset-WS80/31759	{"model":"Fineoffset-WS80","id":31759,"battery_ok":1.11,"battery_mV":3180,"temperature_C":19.7,"humidity":46,"wind_dir_deg":298,"wind_avg_m_s":3,"wind_max_m_s":3.4,"uvi":4,"light_lux":63680,"flags":162,"mic":"CRC","protocol":"Fine Offset Electronics WS80 weather station","rssi":-70,"duration":11001}
```

Sample MQTT event from the WS69:
```
.../Fineoffset-WH65B/211	{"model":"Fineoffset-WH65B","id":211,"battery_ok":1,"temperature_C":19.4,"humidity":47,"wind_dir_deg":249,"wind_avg_m_s":0,"wind_max_m_s":0,"rain_mm":326.136,"uv":2162,"uvi":5,"light_lux":74250,"mic":"CRC","protocol":"Fine Offset Electronics, WH25, WH32B, WH24, WH65B, HP1000, Misol WS2320 Temperature/Humidity/Pressure Sensor","rssi":-79,"duration":11000}
```

Sample data in Home Assistant (note the wind speed is zero because the spinny things are not currently connected as I'm about to mount it in it's real location, but it does work).
<img width="668" alt="image" src="https://github.com/user-attachments/assets/4adfe1a0-f0eb-4a1c-8951-d2e4e729fea6" />

**Reply by neillbell** [reply]
Thanks for letting us know!

**Reply by kbahey** [reply]
Thanks indeed for a comprehensive reply ...

**Comment by sillyfrog** [comment]
I'm considering getting hardware to use this library as well, and have a WS69. I have a HP2560_C display, and it says the sensor is a `WH65/69` - so I'm hoping that the WH65 that's already supported will just work as @tomburke25 suggests if even their own hardware treats them as the same thing.

**Comment by BombusAlpinus** [comment]
i recently switched from a bresser clearview to an ecowitt ws3900/ws69 and i'm not able to receive data from this external unit.
i'm running an nooelec usb dongle and rtl-433 25.02 which works fine with the clearview (-f 868M -R 172) but not with the ws69/wh65b.  i  tried it without any protocol definition and also with the suggested protocol 78.  any ideas? thanks!

this is the output when both units are "sending":

```
rtl_433 -f 868M 
rtl_433 version 25.02 (2025-02-19) inputs file rtl_tcp RTL-SDR SoapySDR

New defaults active, use "-Y classic -s 250k" if you need the old defaults

Found Rafael Micro R820T tuner
[SDR] Using device 0: Realtek, RTL2838UHIDIR, SN: 00000001, "Generic RTL2832U OEM"
Exact sample rate is: 1000000.026491 Hz
Allocating 15 zero-copy buffers
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2025-12-08 09:08:13
model     : Bresser-7in1 id        : 44008
Temperature: 9.0 C       Humidity  : 99            Wind Gust : 0.0 m/s       Wind Speed: 0.0 m/s       Direction : 222           Rain      : 0.0 mm
Light     : 3.052 klx    Light     : 3052.000 lux  UV Index  : 0.0           Battery   : 1             Integrity : CRC
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
```

**Reply by zuckschwerdt** [reply]
Try https://triq.org/rtl_433/ANALYZE.html to get data packets. Check with https://triq.org/pdv3/ and upload a good sample file as zip here if the packet is complete and not clipping (red bars) much, but still not recognized.

**Comment by BombusAlpinus** [comment]
@zuckschwerdt thanks for your fast reply. i'm by far not a radio technican ;) but i will have a look at it.

**Comment by BombusAlpinus** [comment]
@zuckschwerdt  
ok, don't know if this data packet is a good one, got about 130 packets stored ... maybe you can point me in the right direction then. 
[g130_868.3M_1000k.cu8.zip](https://github.com/user-attachments/files/24033966/g130_868.3M_1000k.cu8.zip)

**Reply by zuckschwerdt** [reply]
That signal is not like other FineOffset signals. The timing is slower, 100 µs PCM here vs usually 58 µs PCM.
The signal is clipping and too close to the center. Try with more distance or less gain and use a center frequency of e.g. `-f 868.2M`

**Reply by BombusAlpinus** [reply]
> That signal is not like other FineOffset signals. The timing is slower, 100 µs PCM here vs usually 58 µs PCM. The signal is clipping and too close to the center. Try with more distance or less gain and use a center frequency of e.g. `-f 868.2M`

heres another file. this time with 868.3 frequency. took one with the most samples.
[g128_868.3M_1000k.cu8.zip](https://github.com/user-attachments/files/24051084/g128_868.3M_1000k.cu8.zip)

if this file is also not ok i will try to reduce the distance between sender and receiver then...

**Reply by zuckschwerdt** [reply]
We need a recording at 868.2 MHz, a little offset to the sender. And we might need more distance to get a lower signal.
There are two alternating signals, a longer and weaker signal and then a shorter and stronger signal. It might be a query-response scheme between base station and remote sensor, not sure.

**Reply by BombusAlpinus** [reply]
ok, so here's a file with 868.2M ... did not change the distance. if needed i will try to increase the distance... 
[g192_868.2M_1000k.cu8.zip](https://github.com/user-attachments/files/24060972/g192_868.2M_1000k.cu8.zip)

**Reply by zuckschwerdt** [reply]
That signal is better. The first packet looks like it has a syncword (d395):
`5555 5555 d395 d395 d90a ba0b 5cb7 1d`
The next two packets look to be the same, and they are at 58 µs and have the known 2dd4 syncword:
`aaaaaaaaaaa 2dd4 241c8801ca63000000330000000000acd501ffffff002f835a0`
Not sure what the contents are though.

For the first kind of packet you might want to try
`rtl_433 -f 868.2M -s 1000k -Y minmax -R 0 -X 'n=name,m=FSK_PCM,s=100,l=100,r=3000,preamble=d395d395'`

For the second kind of packet you might want to try
`rtl_433 -f 868.2M -s 1000k -Y minmax -R 0 -X 'n=name,m=FSK_PCM,s=58,l=58,r=3000,preamble=2dd4'`

Each to gather a time series of codes to observe what values change and thus where the data fields are.

**Reply by BombusAlpinus** [reply]
ran the commands for 5 minutes each... results in the attached zip file 
[packet_infos.zip](https://github.com/user-attachments/files/24062271/packet_infos.zip)

**Reply by zuckschwerdt** [reply]
You need to run, preferably the second command, for hours, maybe a day. The codes you got are all the same
`241c8801bb63000000330000000000062001ffffff002fdf4c` -- we need a progression of different codes.

**Reply by BombusAlpinus** [reply]
let the two commands run for 8 hours each... here's the output... 
[rtl433.zip](https://github.com/user-attachments/files/24095744/rtl433.zip)

**Reply by zuckschwerdt** [reply]
The 100µs signal seems boring, mostly the same two codes alternating: `{72}f35fecb19d57ae909f` and `{72}f254d7db0c07ce304c`

The 58µs codes are good, 25 bytes with two checksums at the end: `add_bytes(b, 24) == b[24];` and `crc8(b, 23, 0x31, 0x30) == 0;`
You can now put those codes in [a BitBench](https://triq.org/bitbench/#c=241c5981b06304010033000000001e8e1101ffffff002f96e5&c=241c5981b06304000033000000001ec74901ffffff002f9857&c=241c5981b06303000033000000001e9f2001ffffff002fdf4c&c=241c5981b06302000033000000001e2aaa01ffffff002f5fe0&c=241c5981b063000000330000000014aa1e01ffffff002fb41a&f=8h%208h8h%208h8h%208h8h%208h8h%208h8h%208h8h%208h8h%208h8h%208h8h%208h8h%208h8h%20CHK8h%20CRC8h%20&z=1&cw=4) and try to find the value fields, e.g. try to find the value that changes with temperature.

**Reply by BombusAlpinus** [reply]
ok, thanks for your help. but i have no clue what to do. as already mentioned i'm by far no radio technican... 
but i tried it with gemini:

```
Daten-Set B: Langzeitvergleich (8 Stunden)

**Rohdaten-Pakete für diesen 8-Stunden-Vergleich (`pak2.txt`):**

* [cite_start]**Paket A (Start: 2025-12-10 16:43:16):** `241c5981b06305010033000000000f1ddf80ffffff8017e73c80` [cite: 75959]
* [cite_start]**Paket B (Ende: 2025-12-11 00:42:43):** `241c4d018a63020000330000000000a95901ffffff002f17f60` [cite: 74759]

**Beobachtete Änderungen in den Feldern über 8 Stunden:**

| Byte-Index | Feld | Start-Hex | Ende-Hex | Änderung (Δ) | Bemerkung |
| :---: | :--- | :--- | :--- | :--- | :--- |
| **02 - 03** | **Rolling Code/ID** | `5981` | `4d01` | Stark | Erwartet über lange Zeit |
| **04 - 05** | **Temperatur** | `b063` | `8a63` | Stark | Erwartete Abkühlung über Nacht |
| **06 - 07** | **Wind** | `0501` | `0200` | Stark | Erwartet über lange Zeit |
| **09** | **Konstanter Wert?** | `33` | `33` | Keine | **Bemerkung:** In Set A war dieser Byte 09: `34`. Hier ist er `33`. |
| **13 - 16** | **Wind/Regen** | `0f1ddf80` | `00a95901` | Stark | Erwartet |
| **20** | **Regen/Flags** | `80` | `00` | Stark | Regen war 0.0 mm in Set A. |

```

**Reply by zuckschwerdt** [reply]
Ok, I'll take a look then.

**Reply by zuckschwerdt** [reply, marked as answer]
Ah nice, the Fine Offset / Ecowitt WS69 is basically the WH65B plus 8 bytes extra data. Support added with #3426

**Reply by kbahey** [reply]
Thanks @zuckschwerdt 

But I am now confused ... 

@sillyfrog reported the WS69 working (per [this comment](https://github.com/merbanan/rtl_433/discussions/3059#discussioncomment-13399886)), yet @zuckschwerdt had to add code for the same model to work?

OpenMQTTGateway is based on rtl_433, and should not have independent code for each device. 

What am I missing?

**Reply by zuckschwerdt** [reply]
If the reception is bad enough and packets are truncated then WH65B will work for WS69. But nominal the WS69 packets (`260` bits) are too long and the WH65B decoder will reject them (`> 215`).

**Reply by zuckschwerdt** [reply]
It is possible that @BombusAlpinus's WS69 is a newer model though and that older WS69 have shorter packets.
@kbahey or @sillyfrog do you have an older WS69? Can you upload a zipped .cu8 of that?

**Reply by kbahey** [reply]
Thanks for elaborating on this ... 

No, I don't have a WS69. 

I have a Vevor that does some odd things (every week or two, excessive values are reported for various sensors, such as Lux, humidity, rainfall, ...etc.) 
So I was considering a WS69 to replace the Vevor. 
However, for the time being, I worked around these values.
If they continue to be annoying, I will consider a WS69 again.

**Reply by zuckschwerdt** [reply]
Are those Vevor excessive values specific fixed numbers?

**Reply by kbahey** [reply]
The value for Lux looks suspicious: 

Yesterday, it was 32639 decimal, which is 0x7F7F

But I have seen other values, e.g. 327670
And the workaround in Home Assistant is to exclude values above a certain rational value.

This may coincide with strange values in other measurements.
They are usually spikes or drops from the previous and next value.

For example, relative humidity:

1765765305418000000 95
1765765325340000000 95
1765765345394000000 47 <---
1765765365316000000 95
1765765385368000000 95

Wind Gusts

1765765265312000000 3.33
1765765285365000000 2.67
1765765305418000000 2
1765765325341000000 2
1765765345394000000 86 <---
1765765365316000000 2
1765765385369000000 2.67
1765765405422000000 2.67

Rainfall

1765765325341000000 496.057
1765765345394000000 7853.032 <---
1765765365316000000 496.057

**Reply by zuckschwerdt** [reply]
Yes, that's what I meant. If the (hex) value looks like an error code or overflow then we might be able to filter that. But the random implausible values are best filtered in some application. If you find consistent reproducible error values open an issue.

**Reply by kbahey** [reply]
I could not find a specific value that is consistent through these incidents ...

These spurious values don't seem to affect the display that comes with the Vevor. 
So they are doing their own sanity checks inside its firmware.

WeeWX also does sanity checks very well.

It is in between that it is hard, Home Assistant ...

**Reply by zuckschwerdt** [reply]
You might find out more if you manage to log a "bad" packet, e.g. try `-R 263:v`

**Reply by BombusAlpinus** [reply]
> Ah nice, the Fine Offset / Ecowitt WS69 is basically the WH65B plus 8 bytes extra data. Support added with #3426

do you want me to test this? if yes, how?

**Reply by zuckschwerdt** [reply]
> do you want me to test this?

You can run the nightly binary and see if that works for your WS69. https://github.com/merbanan/rtl_433/releases/tag/nightly

**Reply by BombusAlpinus** [reply]
looks good except for rainfall and uv  (not uv index).
these values are not correct in my opinion. 
haven't found anything comparable within the ui of the base station. a bit off set because i took the screenshot some minutes ago:

<img width="890" height="362" alt="image" src="https://github.com/user-attachments/assets/e193c576-e467-4ee0-aa9f-ff15d7a0ac5f" />


```
/opt/rtl_433_nightly# ./rtl_433 -f 868M
rtl_433 version -128-NOTFOUND branch  at 202512152000 inputs file rtl_tcp RTL-SDR with TLS

New defaults active, use "-Y classic -s 250k" if you need the old defaults

Found Rafael Micro R820T tuner
[SDR] Using device 0: Realtek, RTL2838UHIDIR, SN: 00000001, "Generic RTL2832U OEM"
Exact sample rate is: 1000000.026491 Hz
[R82XX] PLL not locked!
Allocating 15 zero-copy buffers
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2025-12-16 09:08:11
model     : Fineoffset-WS69                        ID        : 28
Battery   : 1            Temperature: -0.4 C       Humidity  : 95 %          Wind direction: 71        Wind speed: 2.7 m/s       Gust speed: 3.6 m/s       Rainfall  : 13.2 mm       UV        : 24
UV Index  : 0            Light     : 3164.0 lux    Integrity : CRC
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2025-12-16 09:08:27
model     : Fineoffset-WS69                        ID        : 28
Battery   : 1            Temperature: -0.4 C       Humidity  : 95 %          Wind direction: 81        Wind speed: 2.4 m/s       Gust speed: 2.5 m/s       Rainfall  : 13.2 mm       UV        : 25
UV Index  : 0            Light     : 3293.0 lux    Integrity : CRC
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2025-12-16 09:08:43
model     : Fineoffset-WS69                        ID        : 28
Battery   : 1            Temperature: -0.4 C       Humidity  : 95 %          Wind direction: 95        Wind speed: 2.9 m/s       Gust speed: 3.6 m/s       Rainfall  : 13.2 mm       UV        : 26
UV Index  : 0            Light     : 3441.0 lux    Integrity : CRC
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2025-12-16 09:08:59
model     : Fineoffset-WS69                        ID        : 28
Battery   : 1            Temperature: -0.4 C       Humidity  : 94 %          Wind direction: 122       Wind speed: 3.2 m/s       Gust speed: 3.6 m/s       Rainfall  : 13.2 mm       UV        : 26
UV Index  : 0            Light     : 3484.0 lux    Integrity : CRC
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2025-12-16 09:08:59
model     : Fineoffset-WS69                        ID        : 28
Battery   : 1            Temperature: -0.4 C       Humidity  : 94 %          Wind direction: 122       Wind speed: 3.2 m/s       Gust speed: 3.6 m/s       Rainfall  : 13.2 mm       UV        : 26
UV Index  : 0            Light     : 3484.0 lux    Integrity : CRC

```

**Reply by zuckschwerdt** [reply]
Thanks! "UV" is the raw value, you want "UV Index", which matches. Rainfall is an absolute value since battery change. You need to take two measurements (e.g. each day) and diff them.

**Reply by BombusAlpinus** [reply]
> Thanks! "UV" is the raw value, you want "UV Index", which matches. Rainfall is an absolute value since battery change. You need to take two measurements (e.g. each day) and diff them.

ok i see... makes sense ;) and i have to say "thanks for your patient support and work"! 
will wait for the release ...

**Reply by Gyvate108** [reply]
> Thanks! "UV" is the raw value, you want "UV Index", which matches. Rainfall is an absolute value since battery change. You need to take two measurements (e.g. each day) and diff them.
1. not since the last battery change but since the last factory reset or hardware reset (cold start from depleted battery and super-capacitor), then the counter is reset
2. there are two hardware editions of the WS69 - older ones with a 6 bit sensor ID, newer ones with a 8 bit sensor ID. The 8 bit sensor ID is needed for the console to process a potential additional pressure sensor (the default sensor is T&H whereas a T&HP sensor is also possible now). The actual code does not consider the pressure option - likely, because they haven't come along such a model and the data will have to be located in the protocol layout. It's bytes 18-20 (count starting from 1) 17 bits (0-16) - Byte 20, Byte 19 and 1st bit (0) of Byte 18.

**Reply by Gyvate108** [reply]
@zuckschwerdt 
I had to correct some numbers - to be considered ! - And added the pressure data location.
 It's bytes 18-20 (count starting from 1) 17 bits (0-16) - Byte 20, Byte 19 and 1st bit (0) of Byte 18.
 Byte 21 is the pressure data checksum.  There are two checksums: 1 is Byte 17 (Bytes 1-16) and 2 is Byte 25 (Bytes 1-24)
 Full payload length: 25 Bytes

**Reply by zuckschwerdt** [reply]
Thanks for the analysis! I'll add the pressure capability to the decoder.

**Reply by Gyvate108** [reply]
@zuckschwerdt 
one more thing: when the pressure module is not present, the value provided will be 0x1FFFF, correct local pressure (ABS) is value/100 in hPa.

**Reply by Gyvate108** [reply]
also, to suggest a correction of some potentially misleading information in the fineoffset.c source code commentaries
The WH65 and WS69 come in three variants: A, B, C, where the characters A, B and C stand for the transmission frequency
A: 868 MHz, B:915 MHz and C (or nothing): 433 MHz.
It was probably named here WH65B because the first decoding attempts took place with the Ambient clone model with transmits at 915MHz and therefore has the B.

**Reply by zuckschwerdt** [reply]
Can you check if #3453 works and shows pressure values?

**Reply by Gyvate108** [reply]
Unfortunately I can't do that.
I don't have the T&HP sensor built into my WS69 as this is still a first generation model without a pluggable T&H sensor.
For the sensor https://shop.ecowitt.com/en-de/products/sht30-temp-hygr-with-protection-filter-plug-in-type-%E5%A4%8D%E5%88%B6
a 3rd generation WS69/WH65 model is needed. (pluggable sensor and a 3-character [24bit encoded] sensor ID - 1st and 2nd generation had a 2-character ID).

The use of a T&HP sensor inside a WS69 is a rather rare use case - together with their new WS6210 WIFI/4G gateway in agriculture and remote locations as the WS6210 doesn't have an inbuilt pressure sensor as almost all FineOffset/Ecowitt consoles have.
It was created as Ecowitt has started targeting agriculture and remote location customers for that new 4G/WIFI [5G/WIFI] gateway with the WS69, WS68, WS80 and WS90 as sensor arrays used.

I have that T&HP sensor inside a WS80 and a WS90 Powered By Shelly, the latter not being useful here as the WS90 PBS transmits via BLE.

The same sensor can be used inside a WS90 and a WS80 where it then provides pressure data.
With the WS90 it's 2 of "your" UU bytes: 6 bits (0-5) of Byte 15 and 8 bits of Byte 16. value/10 in hPa. When no pressure sensor present, the value is 0x3FFF
Byte count starting with 1 (in the fineoffset_WS90.c commentary the count starts with 0).
By the way, the WS90 rain start/stop bit location (bit #4 of Byte 17, bit count starting with 0: the 5th bit) I earlier provided in a PM via wxforum or so.
